src/sessions/yt-music/YTMusicSession.ts

```diff
@@ -83,6 +83,17 @@ export class YTMusicSession implements BrowserSession {
       waitUntil: 'networkidle2',
     });
 
+    // Make sure we have something to click.
+    await this.page.$eval(
+      'yt-formatted-string.text.style-scope.ytmusic-message-renderer', 
+      (element: HTMLSpanElement | HTMLParagraphElement) => {
+        if (element.innerText === 'No results found') new Error('No results found.');
+      }).catch(_ => {
+        // Cancel catch... This should only fire when the element is not found meaning we have song results.
+      });
+    // Make sure we get only songs.
+    await this.page.click('a[title="Show song results"]').catch(_ => new Error('Search element not found.'));
+
```

Comment:
Would it be more concise to use try catch blocks and use the "timeout" option in waitForSelector? Puppeteer, by default, waits 30 seconds before timing out. We can reduce that manually like this:
```
try {
  await page.waitForSelector('a[title="Show song results"]', {
    timeout: 2500, //wait 2.5 seconds before determining that this element doesn't actually exist
  });
} catch (error) {
  //do something here, maybe return out of the function or future code to show main menu
}
```
Just a thought. Your idea may be better because a slow connection might trip the error anyway (ideally not though since we already did a wait for the page to load to 'networkidle2'). But try/catch blocks are my preferred flavor of error handling and could potentially reduce both catch blocks into one here.

Reply:
You're right. The only reason I went this route was so we could know exactly where it breaks when it does. Switching it over now. The shorter timeout is a good idea...